### PR TITLE
Add initial support for enumerated types

### PR DESCRIPTION
**Pull request by elprans**
Enumerated types are a special class of a scalar type that are defined as a
list of explicit string values using the following syntax:

CREATE SCALAR TYPE <name> EXTENDING enum<'val1'[,...'valN']>

Over the wire, and in text serialization, the enum values are transferred 
as strings, but they sort according to the order of enum value definitions, 
and every enum type is distinct, i.e. there is no implicit cast between 
them.

Currently, there is no way to alter the composition of enumerated type 
after it was created.

### CODE REVIEW

File: docs/datamodel/scalars/enum.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    :index: enum
+
+    An enumerated type is a data type consisting of an order list of values.
+
+    An enumeration type can be declared in a schema declaration using
+    the following syntax:
+
+    .. eql:synopsis::
+
+        enum "<" <enum-values> ">"
```
Comment (1st1) [inline docs/datamodel/scalars/enum.rst]: ```scalar type <name> extending enum "<" <enum-values> ">"```

File: docs/edgeql/ddl/scalars.rst [hunk 2]
```diff
@@ -48,6 +48,10 @@ subtype with constraints.
 :eql:synopsis:`EXTENDING <supertype>`
     Optional clause specifying the *supertype* of the new type.
 
+    The :eql:synopsis:`<supertype>` is an
+    :eql:type:`enumerated type <std::enum>` declaration, in which
+    case an enumerated scalar type is defined.
```
Comment (1st1) [inline docs/edgeql/ddl/scalars.rst]: Why are we saying that `<supertupe>` must be an enum declaration in a *generic* section about scalars?

Comment (1st1) [reply docs/edgeql/ddl/scalars.rst]: The wording of this paragraph is weird. Rephrase to something like this: 

To define an enumerated scalar type specify :eql:type:`enumerated type <std::enum>` declaration in :eql:synopsis:`<supertype>` .

Comment (elprans) [reply docs/edgeql/ddl/scalars.rst]: That sentence is missing a leading "if" and was intended like this:

"
    If the :eql:synopsis:`<supertype>` is an
    :eql:type:`enumerated type <std::enum>` declaration, an enumerated scalar type is defined.
"

File: docs/datamodel/scalars/enum.rst (hunk 1 above)
Comment (elprans) [reply docs/datamodel/scalars/enum.rst]: This bit is specifically about the syntax for an anonymous enum type declaration.

**Pull request merged**